src/phoenix/server/api/helpers/playground_clients.py

```diff
@@ -52,6 +54,36 @@
 ChatCompletionChunk: TypeAlias = Union[TextChunk, ToolCallChunk]
 
 
+class KeyedSingleton:
+    _instances: dict[tuple[type, Hashable], Any] = {}
+
+    def __new__(cls, singleton_key: Hashable) -> Any:
+        instance_key = (cls, singleton_key)
+        if instance_key not in cls._instances:
+            instance = super().__new__(cls)
+            cls._instances[instance_key] = instance
+        return cls._instances[instance_key]
+
+
+class PlaygroundRateLimiter(RateLimiter, KeyedSingleton):
+    """
+    A rate rate limiter class that will be instantiated once per `singleton_key`.
+    """
+
+    def __init__(self, singleton_key: Hashable, rate_limit_error: Optional[type[BaseException]]):
```

Comment:
Add default value for `rate_limit_error`?

Reply:
I don't think that's a good idea, it'll throttle and retry and any exception which could lead to very wonky behavior